ast/builtins.go

```diff
@@ -2327,6 +2328,51 @@ with all others being treated as intermediates.`,
 	),
 }
 
+var CryptoX509ParseAndVerifyCertificatesWithOptions = &Builtin{
+	Name: "crypto.x509.parse_and_verify_certificates_with_options",
+	Description: `Returns one or more certificates from the given string containing PEM
+or base64 encoded DER certificates after verifying the supplied certificates form a complete
+certificate chain back to a trusted root. Second argument is a option object to pass
+extra configs to verify the validity of certificates.
+
+The first certificate is treated as the root and the last is treated as the leaf,
+with all others being treated as intermediates.` +
+		"Example option object: " +
+		"`options := {\"DNSName\" : \"example.dns.com\", \"CurrentTime\": 1708447636000000000,`" +
+		"`\"KeyUsages\": {\"KeyUsageServerAuth\", \"KeyUsageClientAuth\", \"KeyUsageCodeSigning\"},`" +
+		"`\"MaxConstraintComparisons\" : 5,`" +
+		"`}`" +
+		"Then this function call will look like: `crypto.x509.parse_and_verify_certificates_with_options(<cert base64 encoded string>, options)`" +
+		"This option's field has same definitions as fields in [x509.VerifyOptions struct](https://pkg.go.dev/crypto/x509#VerifyOptions)" +
+		"`CurrentTime` is nanoseconds since epoch" +
+		"List of possible values for `KeyUsages` field are: `[\"KeyUsageAny\", \"KeyUsageServerAuth\",`" +
+		"`\"KeyUsageClientAuth\", \"KeyUsageCodeSigning\", \"KeyUsageEmailProtection\", \"KeyUsageIPSECEndSystem\",`" +
+		"`\"KeyUsageIPSECTunnel\", \"KeyUsageIPSECUser\", \"KeyUsageTimeStamping\", \"KeyUsageOCSPSigning\",`" +
+		"`\"KeyUsageMicrosoftServerGatedCrypto\", \"KeyUsageNetscapeServerGatedCrypto\",`" +
+		"`\"KeyUsageMicrosoftCommercialCodeSigning\", \"KeyUsageMicrosoftKernelCodeSigning\"]`",
+
+	Decl: types.NewFunction(
+		types.Args(
+			types.Named("certs", types.S).Description("base64 encoded DER or PEM data containing two or more certificates where the first is a root CA, the last is a leaf certificate, and all others are intermediate CAs"),
+			types.Named("options", types.NewObject(
```

Comment:
```suggestion
			types.Named("options", types.NewObject(nil, types.NewDynamicProperty(types.S, types.A)))
```

I think we can just make it a string to any object since there are various types in the struct.